File: encrypter.py
from PIL import Image

# Текст, который нужно закодировать
text = "Альфред докопался до Миши, чтобы тот ему дал поесть печенье"

# Получаем последовательность нулей и единиц для текста
binary_text = ''.join(format(ord(char), 'b').zfill(8) for char in text)

# Размер изображения
image_width = len(binary_text) // 3 + 1
image_height = 3

# Создаем изображение
image = Image.new('RGB', (image_width, image_height), color='white')

# Получаем пиксели изображения
pixels = image.load()

# Заполняем пиксели на основе последовательности нулей и единиц
i = 0
for x in range(image_width):
    for y in range(image_height):
        if i < len(binary_text):
            bit = int(binary_text[i])
            color = (255*bit, 255*bit, 255*bit)
            pixels[x, y] = color
            i += 1

# Сохраняем изображение в файл
image.save('text_image.png')

File: pic-encrypter/Encrypter.cs
﻿using System;
using System.Drawing;
using System.Text;

namespace pic_encrypter
{
    class Encrypter
    {
        static void ConvertToPic(string text)
        {
            string BinaryText = "";
            foreach (char Character in text)
            {
                string BinaryCharacter = Convert.ToString(Character, 2).PadLeft(8, '0');
                BinaryText += BinaryCharacter;
            }

            int ImageWidth = BinaryText.Length / 3 + 1;
            int ImageHeight = 3;

            Bitmap Image = new Bitmap(ImageWidth, ImageHeight);

            for (int YCoordinate = 0; YCoordinate < ImageHeight; YCoordinate++)
            {
                for (int XCoordinate = 0; XCoordinate < ImageWidth; XCoordinate++)
                {
                    Color Color;
                    if (BinaryText.Length > 0)
                    {
                        int Bit = int.Parse(BinaryText.Substring(0, 1));
                        BinaryText = BinaryText.Remove(0, 1);
                        if (Bit == 1)
                        {
                            Color = Color.FromArgb(new Random().Next(256), new Random().Next(256), new Random().Next(256));
                        }
                        else
                        {
                            Color = Color.FromArgb(255, 255, 255);
                        }
                    }
                    else
                    {
                        Color = Color.FromArgb(255, 255, 255);
                    }
                    Image.SetPixel(XCoordinate, YCoordinate, Color);
                }
            }

            Image.Save("text_image.png", System.Drawing.Imaging.ImageFormat.Png);
        }
    }
}
